pandas/tseries/tests/test_resample.py

```diff
@@ -419,25 +419,32 @@ def test_agg_misc(self):
             assert_frame_equal(result, expected, check_like=True)
 
         # series like aggs
-        expected = pd.concat([t['A'].sum(),
-                              t['A'].std()],
-                             axis=1)
-        expected.columns = ['sum', 'std']
-
         for t in [r, g]:
-            result = r['A'].agg({'A': ['sum', 'std']})
+            result = t['A'].agg({'A': ['sum', 'std']})
+            expected = pd.concat([t['A'].sum(),
+                                  t['A'].std()],
+                                 axis=1)
+            expected.columns = ['sum', 'std']
+
+            assert_frame_equal(result, expected, check_like=True)
+
+            expected = pd.concat([t['A'].agg(['sum', 'std']),
+                                  t['A'].agg(['mean', 'std'])],
+                                 axis=1)
+            expected.columns = pd.MultiIndex.from_tuples([('A', 'sum'),
+                                                          ('A', 'std'),
+                                                          ('B', 'mean'),
+                                                          ('B', 'std')])
+            result = t['A'].agg({'A': ['sum', 'std'], 'B': ['mean', 'std']})
```

Comment:
> ahh, but you would be ok with the current if [4] (from 0.18.0), has a multi-level index, right? (e.g. ('C','sum'),('C','std')

yes (so key of dict is always seen as renaming key, independent of length of dict)

>  When shoulud you raise? e.g. just because the key of the dict is not in the columns isn't good enough (or even if its a column in the top-level frame), it HAS to be a renaming key.

If we would want to raise, I think it should not depend on the value of the key (if it is equal to the column name or not), but on the dimension of the values (scalar -> then it is a simple renaming {'my_col_name': my_func} which is OK; when length > 1 -> then it can raise). But indeed probably easier to just allow it, but to be consistent in the handling of the keys as the column names in the multi-index